homeassistant/components/poolsense/binary_sensor.py

```diff
@@ -48,6 +50,6 @@ class PoolSenseBinarySensor(PoolSenseEntity, BinarySensorEntity):
     """Representation of PoolSense binary sensors."""
 
     @property
-    def is_on(self):
+    def is_on(self) -> bool:
         """Return true if the binary sensor is on."""
-        return self.coordinator.data[self.entity_description.key] == "red"
+        return cast(bool, self.coordinator.data[self.entity_description.key] == "red")
```

Comment:
Why is a cast needed here, how can the result of an equality check not be a bool?

Reply:
Based on https://github.com/python/mypy/issues/5697#issuecomment-425738430, comparing with `Any` is always `Any`